src/pages/tags/[tag].page.tsx

```diff
@@ -0,0 +1,80 @@
+import { ParsedUrlQuery } from 'querystring';
+
+import { ReactElement } from 'react';
+
+import { GetStaticPaths, GetStaticProps } from 'next';
+import Head from 'next/head';
+
+import Layout from 'components/Layout';
+import PostList from 'components/Post/List';
+import { getPageTitle } from 'helpers/pageTitle';
+import { getAllPosts, getPostByTag, Post } from 'lib/post';
+
+interface TagProps {
+  tag: string;
+  posts: Post[];
+}
+
+interface PathProps {
+  params: { tag: string };
+}
+
+interface TagParams extends ParsedUrlQuery {
+  tag: string;
+}
+
+export const tagDataTestIds = {
+  heading: 'tag-heading',
+};
+
+const Tag = ({ tag, posts }: TagProps) => {
+  return (
+    <>
+      <Head>
+        <title>{getPageTitle(tag)}</title>
+      </Head>
+
+      <h1
+        className="m-8 text-7xl items-center text-center"
+        data-test-id={tagDataTestIds.heading}
+      >
+        Posts with {tag} tag
+      </h1>
+
+      <PostList posts={posts} itemsRowWise={true} />
+    </>
+  );
+};
+
+Tag.getLayout = (page: ReactElement) => <Layout>{page}</Layout>;
+
+export default Tag;
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const posts = getAllPosts();
+
+  // Get the paths we want to pre-render based on posts
+  let paths: PathProps[] = [];
+  posts.forEach((post) => {
```

Comment:
I think we can move this function into `lib/post.ts` => `getAllTags`? What do you think?

Reply:
I think it's not really fetching all tags to consider it as a separate function `getAllTags`, rather it's building an array of
`params: { tag: string }` which is specific to `getStaticPaths` function, so decided to not move it.